RELEASE/scripts/autoscend.ash

```diff
@@ -2300,57 +2300,58 @@ boolean questOverride()
 
 boolean L13_towerNSNagamar()
 {
+	//status != 12 is an override in case auto_wandOfNagamar is incorrectly set to false
+	//status 11 means ready to fight sorceress, status 12 means you lost to her because you lacked a wand of nagamar
 	if (!get_property("auto_wandOfNagamar").to_boolean() && internalQuestStatus("questL13Final") != 12)
 	{
 		return false;
 	}
-
+	
 	if(item_amount($item[Wand of Nagamar]) > 0)
 	{
 		set_property("auto_wandOfNagamar", false);
 		return false;
 	}
-	else if (internalQuestStatus("questL13Final") == 12)
+	
+	if(creatable_amount($item[Wand Of Nagamar]) > 0)
+	{
+		return create(1, $item[Wand Of Nagamar]);
+	}
+	
+	//hunt for bear verb orgy
+	if (item_amount($item[Wand of Nagamar]) == 0 && internalQuestStatus("questL13Final") == 12)
 	{
 		return autoAdv($location[The VERY Unquiet Garves]);
 	}
-	else if(pulls_remaining() >= 2)
+	
+	if(auto_my_path() == "G-Lover")
 	{
-		if((item_amount($item[ruby w]) > 0) && (item_amount($item[metallic A]) > 0))
-		{
-			cli_execute("make " + $item[WA]);
-		}
-		pullXWhenHaveY($item[WA], 1, 0);
-		pullXWhenHaveY($item[ND], 1, 0);
-		cli_execute("make " + $item[Wand Of Nagamar]);
-		return true;
+		pullXWhenHaveY($item[Ten-Leaf Clover], 1, 0);
 	}
 	else
 	{
-		if(auto_my_path() == "G-Lover")
+		pullXWhenHaveY($item[Disassembled Clover], 1, 0);
+	}
+	
+	if(cloversAvailable() > 0)
+	{
+		cloverUsageInit();
+		autoAdv($location[The Castle in the Clouds in the Sky (Basement)]);
+		cloverUsageFinish();
+		if(creatable_amount($item[Wand Of Nagamar]) > 0)
 		{
-			pullXWhenHaveY($item[Ten-Leaf Clover], 1, 0);
+			return create(1, $item[Wand Of Nagamar]);
 		}
 		else
 		{
-			pullXWhenHaveY($item[Disassembled Clover], 1, 0);
-		}
-		if(in_hardcore() && in_koe())
-		{
-			// TODO: Improve support
-			abort("In Kingdom of Exploathing: Please buy a Wand of Nagamar from the bazaar and re-run.");
-			return false;
-		}
-		else if(cloversAvailable() > 0)
-		{
-			cloverUsageInit();
-			autoAdvBypass(322, $location[The Castle in the Clouds in the Sky (Basement)]);
-			cloverUsageFinish();
-			cli_execute("make " + $item[Wand Of Nagamar]);
-			return true;
+			auto_log_warning("Clovering [The Castle in the Clouds in the Sky (Basement)] for wand parts failed for some reason", "red");
 		}
-		return false;
 	}
+	else if(in_koe())
+	{
+		return buy($coinmaster[Cosmic Ray\'s Bazaar], 1, $item[Wand of Nagamar]);
+	}
```

Comment:
I think you'll need to move this above line 2316 (the first  `if(creatable_amount($item[Wand Of Nagamar]) > 0)` ) otherwise KoE runs might run into issues failing to adventure in a location they can't get to.

Apologies, should've thought of this earlier.

Reply:
I don't understand, why would I want to remove line 2316? it is doing
```
if(creatable_amount($item[Wand Of Nagamar]) > 0)
	{
		return create(1, $item[Wand Of Nagamar]);
	}
```
and not attached to any elses.
if you have the components to make the wand, then it makes the wand.
if you don't it continues to next steps, which include adventuring, clovering, or other things.

Reply:
move above not remove. As in make that else if into it's own little if.

Reply:
there is no else in there

Reply:
the else you highlighted on line 2350 is the else to this if
```if(cloversAvailable() > 0)```
on line 2336

Reply:
Ok, rereading the whole conversation, and the code. I understand now.
Is it the very unqueit garves that is unavailable zone in koe?
wiki says unquiet garves are unavailable but not the very unquiet garves, wiki can be wrong of course.
Or do you mean that the castle basement is unvailable? because wiki doesn't mention that zone as missing.

I think I will just do a koe run to check which of those 2 zones is available, correct the wiki, and then adjust the function here as needed.

Reply:
take the part that says

`else if(in_koe())
	{
		return buy($coinmaster[Cosmic Ray\'s Bazaar], 1, $item[Wand of Nagamar]);
	}`
Delete the word `else`
Then move it to before the first line that says this `if(creatable_amount($item[Wand Of Nagamar]) > 0)`

Why? Well think about what happens in KoE. First time through the function (in a normal run) it will pull a clover for _no reason_ at this line `pullXWhenHaveY($item[Disassembled Clover], 1, 0);` and then fail to adventure in the Castle Basement because it doesn't exist at this line `autoAdv($location[The Castle in the Clouds in the Sky (Basement)]);`
Then the warning that we couldn't create the wand will fire _then_ it will actually buy the wand from Cosmic Ray.

At best, you're wasting a pull. At worst mafia decides to abort because it can't adventure in Castle Basement.

Reply:
Thank you for clarifying, and sorry for that.
I misread your initial post earlier and it set in my mind some assumptions that colored the rest of the conversation. So I completely misunderstood what you were talking about.

In regards to your suggestion here.
My assumption was that it was a good idea to clover in koe because a clover is 1k meat + 1 adv (which I value at 5k meat, so its 6k total) while buying a wand in ray bazaar is 30k meat (unless you have excess isotopes, then its free).
... but yea, in retrospect people will probably complain more about the wasted adventure. So I will do as you suggested.

Aside from that, I would like to koe proof the function some more.
Wiki explicitly states you can clover in koe, I want to verify this.
Wiki explicitly states you can't go to unquiet graves, but does not mention the VERY unquiet graves. And we hunt for bear verb orgy in the latter.

If bear verb orgy is really impossibly in koe then current code might get stuck in an infinite loop and would need some koe proofing.
So in addition to making the change you suggested, I will also do some koe specific spading and then testing.

Reply:
wait, would koe "createable amount" account for ray bazzar? or would it just be crafting it from components?

Reply:
In KoE it should just buy from Cosmic Ray. By the time you get to the tower, you have so many isotopes to spare that not buying it is a waste. Every quest you finish you get 10 isotopes from the council plus you start with 10 or 20 more every run you finish. They're not in short enough supply that you should be spending meat on them and there aren't that many things we buy with them that you'll run out.

```
Quick test of functions on my main account
> ash item_amount(to_item("fat loot token"));

Returned: 91

> ash creatable_amount(to_item("jarlsberg's key"));

Returned: 91

> ash create(1, to_item("jarlsberg's key"));

Verifying ingredients for Jarlsberg's key (1)...
Purchasing Jarlsberg's key (1 @ 1 fat loot token)...
Visiting the Vending Machine...
You acquire an item: Jarlsberg's key
Vending Machine successfully looted!
Successfully created Jarlsberg's key (1)
Returned: true
```
(I needed one for key lime making anyway)
So I think the creatable_amount and create should also handle KoE as it'll check you have the requisite 30 rare meat isotopes and if so, buy the wand from Cosmic Ray. Thanks for bringing this up, I didn't actually think to check if those functions handled coinmasters.

Reply:
I tested and createable amount really does not handle isotope buying from cosmic ray. I had 120 isotopes and it returned 0 for createable amount instead of 4.

I recall not having any isotopes in early runs, but it must have been in very early runs because now i have tons of them.
Anyways, I moved it up as you suggested earlier. just added some checks to avoid errors. the function is only called when you are at the sorceress itself. at that point if you don't have the isotopes you should do a pull. So I put a check for isotope amount.
Also without said check it would have gotten stuck on an infinite loop if we just did return buy(wand)

I can confirm you got access to castle basement. But I am not seeing access to graveyard at all, which means the wiki info was incomplete. ill edit it later. i should also make an exception so it does not try to do verb bear orgy as koe